### PR TITLE
[wundergroundupdatereceiver] Bugfixes: Regenerate trigger channel with proper type and more metadata. Normalize channel names. It might be easiest to delete and allow recreation of channels.

### PR DESCRIPTION
**Pull request by danieldemus**
Bugfix: Replace illegal characters in generated ids with "-"
Bugfix: Not all channel metadata was included on generated channels
Bugfix: The synthetic LAST_QUERY channel was being re-created repetitively

Minor updates to the documentation reflect the changes.

There is an artifact at https://openhab.jfrog.io/artifactory/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.wundergroundupdatereceiver/3.4.0-SNAPSHOT/org.openhab.binding.wundergroundupdatereceiver-3.4.0-SNAPSHOT.jar

### CODE REVIEW

**Comment by danieldemus** [comment]
Fixes #13214

File: bundles/org.openhab.binding.wundergroundupdatereceiver/README.md [hunk 1]
```diff
@@ -42,7 +43,9 @@ If you don't plan on submitting measurements to wunderground.com, it can be any
 Each measurement type the wunderground.com update service accepts has a channel.
 The channels must be named exactly as the request parameter they receive.
 I.e. the wind speed channel must be named `windspeedmph` as that is the request parameter name defined by Wunderground in their API.
-The channel name set up in the binding should be considered an id with no semantic content other than pointing to the wounderground API.
+Illegal channel id characters are converted to -.
+Fx. AqPM2.5 has a channel named `AqPM2-5`.
```
Comment (jlaur) [inline bundles/org.openhab.binding.wundergroundupdatereceiver/README.md:47]:
```suggestion
For example, AqPM2.5 has a channel named `AqPM2-5`.
```

File: pom.xml [hunk 2]
```diff
@@ -127,6 +127,14 @@
     </dependencies>
   </dependencyManagement>
 
+  <pluginRepositories>
+    <pluginRepository>
+      <id>openhab-libs-snapshot</id>
+      <name>Openhab Snapshot Repository</name>
+      <url>https://openhab.jfrog.io/artifactory/libs-snapshot</url>
+    </pluginRepository>
+  </pluginRepositories>
```
Comment (jlaur) [inline pom.xml:136]: Why is this added?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only a minor README and question about POM change. Additionally, would you be able to change the PR title to something which could go into the release notes?

**Comment by s-pw** [comment]
@danieldemus did you test your changes? I've built the binding from [danieldemus:bug-rollup](https://github.com/danieldemus/openhab-addons/tree/bug-rollup) and while the error is gone the channel is gone too.
Adding the channel manually logs error:
Channel type wundergroundupdatereceiver:AqPM2-5 could not be resolved.

File: pom.xml (hunk 2 above)
Comment (danieldemus) [reply pom.xml]: By mistake...

File: bundles/org.openhab.binding.wundergroundupdatereceiver/README.md [hunk 3]
```diff
@@ -42,7 +43,9 @@ If you don't plan on submitting measurements to wunderground.com, it can be any
 Each measurement type the wunderground.com update service accepts has a channel.
 The channels must be named exactly as the request parameter they receive.
 I.e. the wind speed channel must be named `windspeedmph` as that is the request parameter name defined by Wunderground in their API.
-The channel name set up in the binding should be considered an id with no semantic content other than pointing to the wounderground API.
+Illegal channel id characters are converted to -.
+For example AqPM2.5 has a channel named `AqPM2-5`.
```
Comment (jlaur) [inline bundles/org.openhab.binding.wundergroundupdatereceiver/README.md:47]:
```suggestion
For example, AqPM2.5 has a channel named `AqPM2-5`.
```

**Review by jlaur — APPROVED** [review]
Thanks for the fixes.

**Pull request merged**